File: Lv. 2/큰 수 만들기.py
# def solution(number, k):
#     answer = ''    
    
#     l = len(number) - k
    
#     while l != 0:
#         for n in sorted(list(number), reverse=True):
#             if len(number[number.index(n):]) >= l:
#                 answer += n
#                 l -= 1
#                 number = number[number.index(n)+1:]
#                 break        
    
#     return answer

def solution(number, k):    
    
    stack = [number[0]]
    
    for n in number[1:]:
        while k > 0 and stack and stack[-1] < n:
            stack.pop(-1)
            k -= 1
        stack.append(n)
        
    return ''.join(stack[:len(number) - k])

File: Programmers/Lv. 2/큰 수 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    
    vector<char> tmp = {number[0]};
    
    for (int i = 1; i < number.length(); i++)
    {
        while (k > 0 && int(tmp.back()) < int(number[i]) && tmp.size() > 0)
        {
            tmp.pop_back();
            k--;
        }
        tmp.push_back(number[i]);
    }
    
    for (int i = 0; i < tmp.size() - k; i++)
    {
        answer += tmp[i];
    }
    return answer;
}